libsolidity/codegen/ir/IRGenerationContext.h

```diff
@@ -102,7 +102,8 @@ class IRGenerationContext
 
 	std::string newYulVariable();
 
-	std::string internalDispatch(YulArity const& _arity);
+	std::string generateInternalDispatchFunction(YulArity const& _arity);
+	std::string generateInternalDispatchFunction(YulArity const& _arity, std::set<FunctionDefinition const*> _functions);
```

Comment:
I still don't like this split very much. The problem here is that both functions are named the same and calling one function at whatever time is always safe but calling the second function with the wrong second argument can result in severe damage. Is it compatible with what follows to do it as follows?

```
string IRGenerationContext::generateInternalDispatchFunction(YulArity const& _arity)
{
	string funName = IRNames::internalDispatch(_arity);
    set<FunctionDefinition const*> functions = collectFunctionsOfArity(_arity); // this is the second function
	return m_functions.createFunction(funName, [funName, _arity, functions(move(_functions))]() {
        ...
    });
}
```

Reply:
Done.

It's fine for this PR but it forced me to reorganize the code in #8797 in an awkward way. I had to move the whole `generateInternalDispatch()` from `IRGenerator` to `IRGenerationContext` because otherwise I'd have to add more methods to context only to allow it to get the arities without consuming the dispatch content. It doesn't feel quite right to have a whole part of code generation (rather than just helpers) in the context.